src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2932,8 +2974,14 @@ ConnectionStateMachine(WorkerSession *session)
 
 			case MULTI_CONNECTION_CONNECTED:
 			{
-				/* connection is ready, run the transaction state machine */
-				TransactionStateMachine(session);
+				/*
+				 * Connection is ready, and we have unfinished tasks.
```

Comment:
minor: we can move this into if block

Reply:
+1 comment describes the situation inside the if